core/src/main/scala/tech/beshu/ror/utils/SSLCertHelper.scala

```diff
@@ -0,0 +1,195 @@
+import tech.beshu.ror.configuration.SslConfiguration
+import tech.beshu.ror.configuration.SslConfiguration.{KeystoreFile, KeystorePassword, TruststorePassword}
+
+import java.io.{FileInputStream, IOException}
+import java.security.KeyStore
+import javax.net.ssl.{KeyManagerFactory, TrustManagerFactory}
+import scala.collection.JavaConverters._
+import scala.language.{existentials, implicitConversions}
+import scala.util.{Failure, Success, Try}
+
+object SSLCertHelper extends Logging {
+
+  private def getKeyManagerFactoryInstance(fipsCompliant: Boolean) = {
+    if(fipsCompliant) {
+      KeyManagerFactory.getInstance("X509", BouncyCastleJsseProvider.PROVIDER_NAME)
+    } else {
+      KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+    }
+  }
+
+  private def getTrustManagerFactoryInstance(fipsCompliant: Boolean) = {
+    if(fipsCompliant) {
+      TrustManagerFactory.getInstance("X509", BouncyCastleJsseProvider.PROVIDER_NAME)
+    } else {
+      TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
+    }
+  }
+
+  private def loadKeystore(sslConfiguration: SslConfiguration, fipsCompliant: Boolean) = {
+    Resource
+      .fromAutoCloseable(IO(new FileInputStream(sslConfiguration.keystoreFile.value)))
+      .use { keystoreFile => IO {
+        val keystore = if (fipsCompliant) {
+          logger.info("Trying to load keystore in FIPS compliant BCFKS format...")
+          java.security.KeyStore.getInstance("BCFKS", "BCFIPS")
+        } else {
+          logger.info("Trying to load keystore in JKS or PKCS#12 format...")
+          java.security.KeyStore.getInstance("JKS")
+        }
+        keystore.load(keystoreFile, sslConfiguration.keystorePassword)
+        keystore
+      }
+      }
+  }
+
+  private def loadTruststore(sslConfiguration: SslConfiguration, fipsCompliant: Boolean) = {
+    sslConfiguration.truststoreFile
+      .map { truststoreFileName =>
+        Resource
+          .fromAutoCloseable(IO(new FileInputStream(truststoreFileName.value)))
+          .use { truststoreFile =>
+            IO {
+              val keystore = if (fipsCompliant) {
+                logger.info("Trying to load truststore in FIPS compliant BCFKS format...")
+                java.security.KeyStore.getInstance("BCFKS", "BCFIPS")
+              } else {
+                logger.info("Trying to load truststore in JKS or PKCS#12 format...")
+                java.security.KeyStore.getInstance("JKS")
+              }
+              keystore.load(truststoreFile, sslConfiguration.truststorePassword)
+              keystore
+            }
+          }
+      }.sequence
+  }
+
+  private def prepareAlias(keystore: KeyStore, config: SslConfiguration) =
+    config.keyAlias match {
+      case None if keystore.aliases().hasMoreElements =>
+        val firstAlias = keystore.aliases().nextElement()
+        logger.info(s"ROR SSL: ssl.key_alias not configured, took first alias in keystore: $firstAlias")
```

Comment:
warn level?

Reply:
Hmm I don't think it would be good idea to set this log message to warn because probably it's typical situation that alias is not configured